### ISSUE TITLE
Related Posts: Add filter for post excerpt

### ISSUE THREAD

**Issue by chaselivingston** [Automattic/jetpack#3235]
Some themes add shortcodes that aren't registered in the proper way, and thus don't get stripped from the post excerpt in our usual way. We could add a filter [here](https://github.com/Automattic/jetpack/blob/master/modules/related-posts/jetpack-related-posts.php#L775) to help users avoid having tons of shortcode spit out.

**Comment by jeherve** [comment]
See #2884 for more information.

**Comment by jeherve** [comment]
Related: #3278. 

We should probably consider adding a `jetpack_sharing_display_markup` filter that would cover each and every parameter of the Related Posts output. That would solve this issue, as well as others (#2164 and #2220 for example).

**Comment by jeherve** [comment]
Also suggested here:
https://twitter.com/user/status/788471284644585473

**Comment by chaselivingston** [comment]
Update to my original post - if we just want to add a filter for the excerpt content, that would be done [here](https://github.com/Automattic/jetpack/blob/master/modules/related-posts/jetpack-related-posts.php#L941) now.

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/related-posts-how-to-modify-the-title-tag/

**Comment by lamdayap** [comment]
reported here: https://wordpress.org/support/topic/remove-tooltip-on-related-posts-thumbs/

Looks like Visual Composer is adding some shortcode into the title attribute.

**Comment by jeherve** [comment]
Also suggested here: #7718

**Comment by jeherve** [comment]
Also suggested here:
https://wordpress.org/support/topic/change-excerpt-length-5/

**Comment by StefMattana** [comment]
Reported/suggested here: [https://wordpress.org/support/topic/related-post-function-problem-with-qtranslate-x-plugin/](https://wordpress.org/support/topic/related-post-function-problem-with-qtranslate-x-plugin/)

**Comment by jeherve** [comment]
also reported here:
https://wordpress.org/support/topic/disable-wpbakery-visual-composer-code-in-related-posts-image-alt/

**Comment by StefMattana** [comment]
Also reported here: https://wordpress.org/support/topic/how-to-strip-title-attribute-from-related-posts/

**Comment by tmmbecker** [comment]
also reported here: https://wordpress.org/support/topic/jetpack-related-posts-remove-thumbnails-title/

**Comment by jenhooks** [comment]
We have another similar request, here: https://wordpress.org/support/topic/related-posts-remove-post-excerpt/

**Comment by Salubritas** [comment]
I also had a problem with Google indexing post excerpts in the related posts and found a way to remove the excerpts completely, which is to use the filter jetpack_relatedposts_returned_results to remove them from the raw data before it is used to render the results on the page.

`function wr_jetpack_relatedposts_returned_results($results, $post_id) {  `
`    foreach ($results as $key => $subArr) {`
`        $subArr['excerpt'] = '';`
`        $results[$key] = $subArr;  `
`    }`
`    return $results;`
`}`
`add_filter('jetpack_relatedposts_returned_results', 'wr_jetpack_relatedposts_returned_results', 10, 2);`

Hope that helps someone.

**Comment by MuhammedSwalihct** [comment]
Thanks for mentioning it here @jenhooks.

I do not know why Jetpack adds such a huge hidden excerpt on every single related post. I know that it might help some "Accessibility" features for screen readers and such. But the problem is that even though the search engines claim to be smart enough to recognize such things, I do not think they will just ignore around 300-500 words enclosed in <p> tags (if the user has 6 related posts displayed for example) that are the exact beginning sentences of other articles (most people will not put different excerpts for posts.)

For example, let's say someone publishes a news article with 300 words and this excerpt thing exists on that page. It can be a big trouble unless the search engines can determine the purpose of the duplicate content on that page which is even more than the actual content.

Apart from that, even from an Accessibility point of view, I do not think it is a "Must" to have a long description attached to every single related post. We already have titles for that. It doesn't worth risking the rankings of every website that even doesn't know this excerpt is being put on their articles.

Also, thanks to @Salubritas for the code. Will try it.

**Comment by MuhammedSwalihct** [comment]
This still hasn't fixed. And I am having a hard time changing the JS every single time I update the plugin. Dequeue Javascript also doesn't seem to work from Jetpack 9.8 as I noted here. https://github.com/Automattic/jetpack/issues/18736#issuecomment-856451444.

**Comment by cliffordp** [comment]
I'm using v11.9.1 and don't see any excerpts even in the Title (hover)
How can I get the excerpt text to display to the user under the image without requiring hover (i.e. only in the Title)?

**Comment by jeherve** [comment]
> I'm using v11.9.1 and don't see any excerpts even in the Title (hover)

That's correct. We removed excerpts from the post title a few years ago, in #16088, to improve accessibility.

> How can I get the excerpt text to display to the user under the image without requiring hover (i.e. only in the Title)?

That's currently not possible using the built-in implementation (either with the block or the layout added to the bottom of the posts).

You could, however, develop your own Related Posts markup using the `Jetpack_RelatedPosts_Raw` class. You can find out more about this here:
https://jetpack.com/support/related-posts/customize-related-posts/#raw

**Comment by cliffordp** [comment]
@jeherve tyvm for the detailed reply. Can you log a feature request to display excerpts via UI settings?

**Comment by jeherve** [comment]
Sure thing! I've just created #29727 to keep track of that.
